### PR TITLE
Add Tencent Hy-MT2-1.8B support (L4/L5 golden + parity example)

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

Adds support + verification for [`tencent/Hy-MT2-1.8B`](https://huggingface.co/tencent/Hy-MT2-1.8B), the newer (2026.5) multilingual translation checkpoint in the **HunYuanDenseV1** family (`model_type: hunyuan_v1_dense`).

The architecture is **already supported** by `HunYuanV1DenseCausalLMModel` (GQA 16/4 heads, head_dim 128, per-head QK-norm, tied embeddings, dynamic-NTK RoPE with `alpha=1000` via `DynamicNTKRope`). No model/registry code changes were needed — this PR adds the test coverage and a runnable demo that prove it works end-to-end on the real weights.

## Changes

- **`testdata/cases/causal-lm/hy-mt2-1_8b.yaml`** — L4+L5 case, pinned to the model revision, with the Hy-MT chat-template-formatted translation prompt.
- **`testdata/golden/causal-lm/hy-mt2-1_8b.json`** + **`...-1_8b_generation.json`** — L4 prefill-logits and L5 greedy-generation goldens (generated against HF `transformers`).
- **`examples/hy_mt2.py`** — builds the ONNX export, runs greedy generation through **ONNX Runtime** *and* **HuggingFace**, and asserts the two produce **identical** token sequences.

## Validation

- `pytest tests/e2e_golden_test.py -m "golden or generation" -k hy_mt2` → **2 passed** (L4 prefill logits + L5 greedy generation match HF).
- `examples/hy_mt2.py` reproduces HF token-for-token:
  ```
  source : 黄河之水天上来
  ONNX   : The waters of the Yellow River come from the sky
  HF     : The waters of the Yellow River come from the sky
  ✅ PASS: ONNX and HuggingFace produced identical 11 tokens.
  ```
- `hunyuan_v1_dense` unit tests pass; lint clean.

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

### CODE REVIEW

File: examples/hy_mt2.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    # Exact, portable parity on CPU (float32 both sides):
+    python examples/hy_mt2.py
+
+    # Custom translation prompt:
+    python examples/hy_mt2.py --source "今天天气真好。" --target-lang English
+
+    # Half precision on CUDA (bf16 MatMul is unsupported on the ORT CPU EP):
+    python examples/hy_mt2.py --dtype f16 --device cuda
+
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply examples/hy_mt2.py:25]: Fixed in 9008d6d. Rather than restricting the demo to f32, I made `OnnxGenerator` dtype-aware: it now resolves each `past_key_values.*` / recurrent-state input dtype via `session.get_input_dtype` (which maps bf16 through ml_dtypes) and allocates the zero KV buffers to match, falling back to float32 when unknown. Validated end-to-end: `examples/hy_mt2.py --dtype f16 --device cuda` now runs and matches HuggingFace token-for-token; the f32 path is unchanged.

File: examples/hy_mt2.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        "--device",
+        choices=("cpu", "cuda"),
+        default="cpu",
+        help="ORT/torch device. f16/bf16 require cuda (no CPU MatMul kernels).",
+    )
+    parser.add_argument("--max-new-tokens", type=int, default=32)
+    args = parser.parse_args()
+
+    if args.dtype in ("f16", "bf16") and args.device != "cuda":
+        parser.error(f"--dtype {args.dtype} needs --device cuda (no ORT CPU MatMul kernel)")
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply examples/hy_mt2.py:114]: Fixed in 9008d6d. Rather than restricting the demo to f32, I made `OnnxGenerator` dtype-aware: it now resolves each `past_key_values.*` / recurrent-state input dtype via `session.get_input_dtype` (which maps bf16 through ml_dtypes) and allocates the zero KV buffers to match, falling back to float32 when unknown. Validated end-to-end: `examples/hy_mt2.py --dtype f16 --device cuda` now runs and matches HuggingFace token-for-token; the f32 path is unchanged.
